interface.go

```diff
@@ -211,6 +211,18 @@ type EarlyConnection interface {
 	NextConnection() Connection
 }
 
+// A ConnectionIDGenerator is a interface that allows clients to implement their own format
+// for the Connection IDs that servers/clients use as SrcConnectionID in QUIC packets.
+type ConnectionIDGenerator interface {
+
+	// GenerateConnectionID generates a new ConnectionID.
```

Comment:
```suggestion
	// GenerateConnectionID generates a new Connection ID.
```

Can you also add some text to clarify that Connection IDs need to unique, and must not allow an observer to correlate them.